general/config.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2010, Lawrence Livermore National Security, LLC. Produced at
+// the Lawrence Livermore National Laboratory. LLNL-CODE-443211. All Rights
+// reserved. See file COPYRIGHT for details.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability see http://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the GNU Lesser General Public License (as published by the Free
+// Software Foundation) version 2.1 dated February 1999.
+
+#include "../general/okina.hpp"
+
+// *****************************************************************************
+namespace mfem
+{
+
+// *****************************************************************************
+// * cudaDeviceSetup will set: gpu_count, dev, cuDevice, cuContext & cuStream
+// *****************************************************************************
+void config::CudaDeviceSetup(const int device)
+{
+#ifdef __NVCC__
+   cudaGetDeviceCount(&ngpu);
+   MFEM_ASSERT(gpu_count>0, "No CUDA device found!");
```

Comment:
Should gpu_count be ngpu? I hit a build error since gpu_count was not defined.

Reply:
Yes, you're right, I'll fix it, thanks Arturo!